Packs/CiscoUmbrellaReporting/Integrations/CiscoUmbrellaReporting/CiscoUmbrellaReporting.py

```diff
@@ -0,0 +1,1317 @@
+        raise DemistoException(PAGE_SIZE_ERROR_MSG)
+
+    page = DEFAULT_OFFSET if not page else page - 1
+    page_size = DEFAULT_PAGE_SIZE if not page_size else page_size
+    limit = page_size
+    offset = page * page_size
+    return limit, offset
+
+
+def create_cisco_umbrella_args(limit: Optional[int], offset: Optional[int], args: Dict) -> Dict:
+    """
+    This function creates a dictionary of the arguments sent to the Cisco Umbrella API based on the demisto.args().
+    Args:
+        limit: Records per page.
+        offset: The number of records to be skipped.
+        args: demisto.args()
+    Returns:
+        Return arguments dict.
+    """
+    cisco_umbrella_args: Dict = {}
+
+    if sha256 := args.get('sha256'):
+        check_valid_indicator_value('sha256', sha256)
+    if ip := args.get('ip'):
+        check_valid_indicator_value('ip', ip)
+    if domains := args.get('domains'):
+        check_valid_indicator_value('domains', domains)
+    if urls := args.get('urls'):
+        check_valid_indicator_value('urls', urls)
+    if intrusion_action := args.get('intrusion_action'):
+        check_valid_indicator_value('intrusion_action', intrusion_action)
+
+    max_limit = arg_to_number(args.get('limit', DEFAULT_PAGE_SIZE), arg_name='limit')
+
+    cisco_umbrella_args['limit'] = limit if limit != DEFAULT_PAGE_SIZE else max_limit
+    cisco_umbrella_args['offset'] = offset
+    cisco_umbrella_args['from'] = args.get('from', DEFAULT_FROM_DATE)
+    cisco_umbrella_args['to'] = args.get('to', DEFAULT_TO_DATE)
+    cisco_umbrella_args['threattypes'] = args.get('threat_types')
+    cisco_umbrella_args['identitytypes'] = args.get('identity_types')
+    cisco_umbrella_args['ampdisposition'] = args.get('amp_disposition')
+    cisco_umbrella_args['filename'] = args.get('file_name')
+    cisco_umbrella_args['intrusionaction'] = intrusion_action
+    cisco_umbrella_args['domains'] = domains
+    cisco_umbrella_args['urls'] = urls
+    cisco_umbrella_args['ip'] = ip
+    cisco_umbrella_args['ports'] = args.get('ports')
+    cisco_umbrella_args['verdict'] = args.get('verdict')
+    cisco_umbrella_args['threats'] = args.get('threats')
+    cisco_umbrella_args['signatures'] = args.get('signatures')
+    cisco_umbrella_args['sha256'] = sha256
+
+    return cisco_umbrella_args
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client) -> str:
+    """
+    Tests API connectivity and authentication
+    When 'ok' is returned it indicates the integration works like
+    it is supposed to and connection to the service is successful.
+    Args:
+        client(Client): Client class object
+    Returns:
+        Connection ok
+    """
+    token = client.access_token()
+    url = f'{client._base_url}/v2/organizations/{client.organisation_id}/activity'
+
+    payload: Dict = {}
+    params: Dict = {
+        'limit': 1,
+        'from': '-1days',
+        'to': 'now',
+        'offset': 0
+    }
+    response = requests.get(
```

Comment:
Please use here the `self._http_request()` instead of calling requests.get directly.

Reply:
@ShacharKidor Done

Reply:
nice :)